scripts/zones/Aht_Urhgan_Whitegate/npcs/Kaduru-Haiduru.lua

```diff
@@ -0,0 +1,125 @@
+-----------------------------------
+-- Area: Aht Urhgan Whitegate
+-- NPC: Kaduru-Haiduru
+-- Teleport NPC
+-----------------------------------
+
+-----------------------------------
+-- onTrigger
+-----------------------------------
+function onTrigger(player, npc)
+    local caughtUsingShihuDanhuDate = player:getVar("Kaduru_ShihuDanhu_date");
+    local shihuDanhuDate = player:getVar("ShihuDanhu_TP_date");
+    local timesUsed = player:getVar("Kaduru_TimesUsed");
+    local currentDate = os.date("%j");
+
+    if ((canUse_KaduruHaiduru_Service(player) == 1)) then
```

Comment:
`if canUse_KaduruHaiduru_Service(player) == 1 then`

..actually, you could modify this to return true/false instead of 1/0, it'd make more sense

Reply:
For this one would it be best to just leave it canUse_KaduruHaiduru_Service(Player) or add the == true?

Reply:
ok confirmed now it does work how I was thinking, we can just say 
```lua
if canUse_KaduruHaiduru_Service(player) then
```
and it'll pass the check only when the `canUse_KaduruHaiduru_Service` function returns `true`